### PR TITLE
Add test suite for FHE contract

### PR DESCRIPTION
**Pull request by ryanorendorff**
Adds a test suite for the FHE contract functions that can run against a live contract. To run locally with anvil for example, you can do:

```
# In one terminal
cargo run --release --manifest-path=crates/fhe-contract-tests/Cargo.toml -- node

# In another terminal
cargo run --release --manifest-path=crates/fhe-contract-tests/Cargo.toml -- test
```

To test against a live endpoint you can also provide the RPC URL endpoint with `-e` and the contract address endpoint with `-a`.

Another PR will add the feature to load a secret key in order to choose a different account. Currently this chooses the test account, which may not exist at a given endpoint.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,3 +53,17 @@ jobs:
...
+      - name: Check committed FHE contract binding
+        run: |
+          forge bind \
+            --bindings-path ./crates/fhe-contract-tests/src/bindings \
+            --module --single-file \
+            --select "^FHE\$"
+
+      - name: Run FHE contract on anvil tests
+        run: |
+          cargo run --release --manifest-path=crates/fhe-contract-tests/Cargo.toml -- node &
```
Comment (samtay) [inline .github/workflows/test.yml]: More typical usage would be `cargo run -p fhe-contract-tests`, but if this works that's fine

File: src/FHE.sol [hunk 2]
```diff
@@ -3,11 +3,9 @@ pragma solidity ^0.8.19;
 
 import "./Bytes.sol";
 
-contract FHE {
```
Comment (samtay) [inline src/FHE.sol]: Did you mean to make this change? This has downstream consequences, and there are a number of PRs currently open that address them. I'd probably revert this change, unless we want to merge all those PRs first.

**Review by samtay — APPROVED** [review]
Approve pending resolution on library contract

File: src/FHE.sol (hunk 2 above)
Comment (ryanorendorff) [reply src/FHE.sol]: Oh whoops no I did not.

File: .github/workflows/test.yml (hunk 1 above)
Comment (ryanorendorff) [reply .github/workflows/test.yml]: Fixed.

File: src/FHE.sol (hunk 2 above)
Comment (ryanorendorff) [reply src/FHE.sol]: Fixed.
